pkg/storegateway/bucket.go

```diff
@@ -626,9 +626,18 @@ func blockSeries(
 		var (
 			symbolizedLset []symbolizedLabel
 			chks           []chunks.Meta
+			postingsStats  = &queryStats{}
 		)
+
+		// Keep track of postings lookup stats in a dedicated stats structure that doesn't require lock
+		// and then merge it once done. We do it to avoid a lock of lock contention because loadSeriesForTime()
```

Comment:
```suggestion
		// and then merge it once done. We do it to avoid a lot of lock contention because loadSeriesForTime()
```

Reply:
does lock contention also include just acquiring the lock without racing with other goroutines? Because the for loop below runs serially, so it wouldn't race with other goroutines

Reply:
Will rephrase. It's not lock contention, but lock overhead anyway.

Reply:
See: https://github.com/grafana/mimir/pull/3399/commits/0b2ac1b11a1f42589289ace9d59f7b4378e7e8f6